pkgs/build-support/testers/default.nix

```diff
@@ -123,7 +123,18 @@
           inherit pkgs;
           extraConfigurations = [(
             { lib, ... }: {
-              config.nixpkgs.pkgs = lib.mkDefault pkgs;
+              # Carefully note that we intentionally don't do this:
+              #
+              # config.nixpkgs.pkgs = lib.mkDefault pkgs;
+              #
+              # … because we want to to be able to reinstantiate `nixpkgs.pkgs`
+              # with the (possibly different) `nixpkgs.system` specified in
+              # `./nixos/lib/testing/nodes.nix`.  This means that only package
+              # overrides specified in the `overlays` or
+              # `config.packageOverrides` arguments to Nixpkgs will be passed
+              # through to `nixpkgs.pkgs`.
+              config.nixpkgs.overlays = lib.mkDefault pkgs.overlays;
+              config.nixpkgs.config = lib.mkDefault pkgs.config;
```

Comment:
This was too much of a breaking change, regressing evaluation performance and possibly other things, considering that this part is heavily, thoroughly used.

Fortunately, we can don't have to pick the same solution in both places. I'm sending a few commits that bring this closer to the previous solution and fix some other small problems.